autogen/main/cluster.tf.tmpl

```diff
@@ -288,6 +288,13 @@ resource "google_container_cluster" "primary" {
 
       service_account = lookup(var.node_pools[0], "service_account", local.service_account)
 
+      tags = concat(
+        lookup(local.node_pools_tags, "default_values", [true, true])[0] ? [local.cluster_network_tag] : [],
+        lookup(local.node_pools_tags, "default_values", [true, true])[1] ? ["${local.cluster_network_tag}-default-pool"] : [],
+        lookup(local.node_pools_tags, "all", []),
+        lookup(local.node_pools_tags, "default-pool", []),
```

Comment:
Thanks for working on this!

> I deviated slightly from @morgante's suggestion in the related issue (adding the network tags from the first pool in the cluster config like for the service account): node_pools_tags is a map-of-lists unlike node_pools, which is a list-of-maps. The "first" pool here wouldn't be well defined -- I think the approach I went with is more natural in any case.

I'd like to avoid introducing a new "default-pool" concept because we already have precedent for using the first node pool as the default pool ([example](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-private-cluster/cluster.tf#L97)). In this case, what we should do is retrieve the name of the first node pool (from `var.node_pools`) and use the network tags associated with that node pool (in `var.node_pools_tags`).

Reply:
Ah, I see -- makes sense! I've changed the `"default-pool"` lookup to `var.node_pools[0].name` -- from what I can see
* node_pools[0] should always exist
* node_pools entries are assumed to have the attribute `name`

so I have no error checking for the cases where either of these aren't true. Let me know if either is not the case.

Did we want to keep the `"${local.cluster_network_tag}-default-pool"` tag?

Reply:
Looks good, yeah you shouldn't need error checking.

> Did we want to keep the "${local.cluster_network_tag}-default-pool" tag?

Yes, I think it's helpful.